internal/html/scrape/schema/json-ld/recipe.go

```diff
@@ -110,11 +136,38 @@ func addSchemaCtx(v any) {
 func findRecipeNode(nodes []any) (map[string]any, bool) {
 	for _, node := range nodes {
 		if m, ok := node.(map[string]any); ok {
-			if m[typeKey] == recipeType {
+			str, arr, err := ConvertToStringOrArray(m[typeKey])
+			if err != nil {
+				return nil, false
+			}
+
+			if arr != nil && ArrayContains(arr, recipeType) {
+				return m, true
+			} else if str != "" && str == recipeType {
 				return m, true
```

Comment:
I think the way you had it before was easier to read. Generally not a fan of returning > 2 variables.

Reply:
I think in that case I will just remove the method entirely, as it is only used here and in the test